docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/json_data_converter_HOWTO.md

```diff
@@ -0,0 +1,330 @@
+
+### 3.2 data_location
+The `data_location` field specifies where the main data array/object is located in your JSON structure. This is typically the root of your telemetry data.
+
+Example:
+```yaml
+data_location: resource_spans
+```
+
+### 3.3 header
+The header section defines paths to important metadata that applies to all spans within a group. It uses a list of paths to locate this information:
+
+```yaml
+header:
+    paths: [resource:attributes, scope_spans::scope:name] 
+```
+* resource:attributes - Metadata about the resource (e.g., service name and version)
+* scope_spans::scope:name - The name of the scope for the spans
+
+This header information can be referenced in the field_mapping section using the `HEADER:` prefix. 
+
+### 3.4 span_mapping
+
+The span_mapping section defines where to find the individual span data within the JSON structure. In this configuration:
+
+```yaml
+span_mapping:
+  spans:
+    key_paths: [scope_spans::spans]
+```
+
+This indicates that the span data is located in the "spans" array within the "scope_spans" array.
+
+NB. spans must always be located within an array, or an error will be thrown.
+
+### 3.5 field_mapping
+
+The `field_mapping` section is the core of the configuration, defining how to map fields from the input JSON to your desired output structure. Each field in your target schema is defined here. For example:
+
+```yaml
+field_mapping:
+    event_type:
+        key_paths: [attributes::key, attributes::key]
+        key_value: [http.method, http.response]
+        value_paths: [value:Value:StringValue, value:Value:IntValue]
+        value_type: string
+```
+
+Returns "GET 200" from the JSON example.
+
+#### The two required fields are `key_paths` and `value_type`.
+
+This mapping for `event_type`:
+
+1. Loops over two locations (using `key_paths`).
+
+2. The key_path identifies the location of the nested key that will identify the value we are searching for.
+
+3. If the `key_value` attribute is provided, that tells us the structure of the data looks like:
+
+```json
+{
+    "key": "http.method",
+    "value": {
+        "Value": {
+            "StringValue": "GET"
+        }
+    }
+}
+```
+as opposed to:
+```json
+{
+    "key": "value"
+}
+```
+
+
+4. Checks if the `key`'s value matches `http.method` in the first path.
+
+3. Retrieves the value from the specified `value_paths`. For the first path it's specified as `value:Value:StringValue`, giving us a value of "GET".
+
+4. Defines the `value_type` as a `string`
+
+5. Results of the second path are concatenated to the first, giving a final output of "GET 200".
+
+Each field in the field_mapping section follows a similar structure, allowing you to precisely define how to extract and transform data from the input JSON to your desired output format.
```

Comment:
"your desired output format" - the output is used internally by the tool and is not for the user